files/scripts/syncd.sh

```diff
@@ -104,11 +104,12 @@ start() {
         if [[ x"$WARM_BOOT" != x"true" ]]; then
             if [[ x"$(/bin/systemctl is-active pmon)" == x"active" ]]; then
```

Comment:
@stephenxs since you have removed pmon restart from CLI, we do not need this anymore.

Reply:
@stephenxs i suggest the next flow:

```bash
if [[ x"$WARM_BOOT" != x"true" ]]; then
    /usr/bin/hw-management.sh chipdown
fi

/bin/systemctl restart pmon

if [[ x"$BOOT_TYPE" == x"fast" ]]; then
    /usr/bin/hw-management.sh chipupdis
fi
```

Reply:
You're correct. The original code prevents pmon from starting in warm-reboot flow.

Reply:
> @stephenxs since you have removed pmon restart from CLI, we do not need this anymore.

I am also considering to remove it. originally I intend to make a protection here so that if pmon is still alive after syncd died somehow we can make it sure that pmon is stopped to avoid racing condition.  however it seems that this situation has never happened. 
so if all of us think it's unnecessary I am going to remove it.

Reply:
> > originally I intend to make a protection here so that if pmon is still alive after syncd died somehow we can make it sure that pmon is stopped to avoid racing condition.
> 
> @stephenxs nice catch. This also should be taken into consideration.

To address this situation, we have two options,
1. to remain the logic that stops pmon if it was active ahead of syncd starting. As I mentioned before, it's also risky even the possibility is very low. If we intend to address this issue, we have:
2. [[syncd.sh,pmon.service] Prevent pmon from starting ahead of syncd](https://github.com/stephenxs/sonic-buildimage/pull/8)

Reply:
> @stephenxs the flow becomes very complicated and some aspects are missing comparing to original config reload. Maybe we should revisit the architecture?

Having gone through all the flow, the difference compared to the original flow includes:
    for config reload and swss restart, pmon stops ahead of syncd in the updated flow and after syncd in the original one.
    for warm-reboot, in the original flow, the point when pmon starts can be one of the following 2 cases:
    1. ahead of syncd started. in this case, the updated flow doesn't have any difference.
    2. at any point after syncd. in this case, in the updated flow the pmon starts immediate after "chipdown" called, which means it may introduce a bit more latency for warm reboot. It can be demonstrated as the following (The "ORI FLOW" stands for the time sequence in the original flow and the "UDT FLOW" stands for that of the updated one):
ORI FLOW:  --syncd starting------chip down--------syncd fully started-----pmon starting------
UDT FLOW: --syncd starting------chip down--pmon starting---syncd fully started-------------
Fortunately, per my test it is always the first case which means no extra latency introduced.